docs/resources/vcpu_server.md

```diff
@@ -0,0 +1,142 @@
+    public                = true
+    name                  = "Lan Example"
+}
+
+resource "ionoscloud_ipblock" "example" {
+    location              = ionoscloud_datacenter.example.location
+    size                  = 4
+    name                  = "IP Block Example"
+}
+
+resource "ionoscloud_vcpu_server" "example" {
+    name                  = "VCPU Server Example"
+    datacenter_id         = ionoscloud_datacenter.example.id
+    cores                 = 1
+    ram                   = 1024
+    availability_zone     = "ZONE_1"
+    image_name            = data.ionoscloud_image.example.id
+    image_password        = random_password.server_image_password.result
+    volume {
+        name              = "system"
+        size              = 5
+        disk_type         = "SSD Standard"
+        user_data         = "foo"
+        bus               = "VIRTIO"
+        availability_zone = "ZONE_1"
+    }
+    nic {
+        lan               = ionoscloud_lan.example.id
+        name              = "system"
+        dhcp              = true
+        firewall_active   = true
+        firewall_type     = "BIDIRECTIONAL"
+        ips               = [ ionoscloud_ipblock.example.ips[0], ionoscloud_ipblock.example.ips[1] ]
+        firewall {
+          protocol          = "TCP"
+          name              = "SSH"
+          port_range_start  = 22
+          port_range_end    = 22
+          source_mac        = "00:0a:95:9d:68:17"
+          source_ip         = ionoscloud_ipblock.example.ips[2]
+          target_ip         = ionoscloud_ipblock.example.ips[3]
+          type              = "EGRESS" 
+        }
+    }
+    label {
+        key = "labelkey1"
+        value = "labelvalue1"
+    }
+    label {
+        key = "labelkey2"
+        value = "labelvalue2"
+    }
+}
+resource "random_password" "server_image_password" {
+  length           = 16
+  special          = false
+}
+                  
+```
+
+## Argument reference
+
+- `name` - (Required)[string] The name of the server.
+- `datacenter_id` - (Required)[string] The ID of a Virtual Data Center.
+- `cores` - (Optional)[integer] Number of server CPU cores.
+- `ram` - (Optional)[integer] The amount of memory for the server in MB.
+- `image_name` - (Optional)[string] The name, ID or alias of the image. May also be a snapshot ID. It is required if `licence_type` is not provided. Attribute is immutable.
+- `availability_zone` - (Optional)[string] The availability zone in which the server should exist. E.g: `AUTO`, `ZONE_1`, `ZONE_2`. This property is immutable.
+- `licence_type` - (Optional)[string] Sets the OS type of the server.
+- `volume` - (Required) See the [Volume](volume.md) section.
+- `nic` - (Optional) See the [Nic](nic.md) section.
+- `firewall` - (Optional) Allows to define firewall rules inline in the server. See the [Firewall](firewall.md) section.
+- `boot_volume` - (Computed) The associated boot volume.
+- `boot_cdrom` - (Optional)[string] The associated boot drive, if any.
```

Comment:
add - (Optional)[string] The associated boot drive, if any. Must be the UUID of a bootable CDROM image that can be retrieved using the [ionoscloud_image](../data-sources/image.md) data source.

Reply:
@cristiGuranIonos done in 52c9a9c839f1d083097f21fbd6777ca8c3fbfb14